### ISSUE TITLE
cannot update because "No such metadata object"

### ISSUE THREAD

**Issue by aleixpol** [flatpak/flatpak#725]
I've been seeing these error messages lately. I'm not really sure why or how, but it looks like an issue in flatpak (or flatpak-builder) since it happens both on the binaries I get from gnome as well as from kde.

Also sometimes it's an error and sometimes it's a warning, although it seems to be the same issue. Running it again doesn't change the output.
```
Warning: Failed to update org.gnome.Sdk/x86_64/3.24: While pulling runtime/org.gnome.Sdk/x86_64/3.24 from remote gnome: No such metadata object 3da9ccd38b65872eb305468abee012b6fe609fa1dbd8234df2c1fa5e38b61ff9.commit
Warning: Failed to update org.kde.Platform/x86_64/master: While pulling runtime/org.kde.Platform/x86_64/master from remote kde: No such metadata object b071891e69fc06dd3d5be595f7249f81c767c9aa3a12d13326901b06abfc4ff7.commit
Error: Failed to update org.kde.kdevelop/x86_64/master: While pulling app/org.kde.kdevelop/x86_64/master from remote kdeapps: No such metadata object 9e075f3153311f2444f9b3e24a2a9a270836a4d1707d0ff82c2f2805baad9ec2.commit
Error: Failed to update org.freedesktop.Platform/x86_64/1.6: While pulling runtime/org.freedesktop.Platform/x86_64/1.6 from remote gnome: No such metadata object 06c485e4db54cc7142af6269232d8a1dc6d60d42c1f4d101fb948087f2e065f8.commit
```

**Comment by alexlarsson** [comment]
Can you run flatpak update -v --ostree-verbose and attach some logs of when this happens.

Also, is this a system or per-user installation?

**Comment by alexlarsson** [comment]
Can you run flatpak update -v --ostree-verbose and attach some logs of when this happens.

Also, is this a system or per-user installation?

**Comment by aleixpol** [comment]
Not sure if it's very helpful but there it goes:
```
$ flatpak update -v --ostree-verbose
Looking for updates...
XA: Updating dependent runtime runtime/org.gnome.Sdk/x86_64/3.24
XA: Transaction: update gnome:runtime/org.gnome.Sdk/x86_64/3.24[$old]
XA: Transaction: install/update gnome:runtime/org.gnome.Sdk.Locale/x86_64/3.24[/en]
XA: Transaction: update gnome-apps:app/org.gnome.Builder/x86_64/stable[$old]
XA: Transaction: install/update gnome-apps:runtime/org.gnome.Builder.Locale/x86_64/stable[/en]
XA: Updating dependent runtime runtime/org.gnome.Platform/x86_64/3.22
XA: Transaction: update gnome:runtime/org.gnome.Platform/x86_64/3.22[$old]
XA: Transaction: install/update gnome:runtime/org.gnome.Platform.Locale/x86_64/3.22[/en]
XA: Transaction: update gnome-apps:app/org.gnome.Geary/x86_64/stable[$old]
XA: Transaction: install/update gnome-apps:runtime/org.gnome.Geary.Locale/x86_64/stable[/en]
XA: Updating dependent runtime runtime/org.kde.Platform/x86_64/master
XA: Transaction: update kde:runtime/org.kde.Platform/x86_64/master[$old]
XA: Transaction: install/update kde:runtime/org.kde.Platform.Locale/x86_64/master[/en]
XA: Transaction: update kdeapps:app/org.kde.katomic/x86_64/master[$old]
XA: Transaction: update kdelocal:app/org.kde.kdeconnect/x86_64/master[$old]
XA: Transaction: update kdeapps:app/org.kde.kpat/x86_64/master[$old]
XA: Transaction: update kdeapps:app/org.kde.kube/x86_64/master[$old]
XA: Transaction: update org.telegram.desktop-origin:app/org.telegram.desktop/x86_64/master[$old]
XA: Transaction: update gnome:runtime/org.freedesktop.Sdk.Locale/x86_64/1.6[$old]
XA: Transaction: update gnome:runtime/org.freedesktop.Sdk/x86_64/1.6[$old]
XA: Transaction: install/update gnome:runtime/org.freedesktop.Sdk.Locale/x86_64/1.6[/en]
XA: Transaction: update gnome-apps:runtime/org.gnome.Builder.Locale/x86_64/stable[$old]
XA: Transaction: update gnome-apps:runtime/org.gnome.Geary.Locale/x86_64/stable[$old]
XA: Transaction: update gnome:runtime/org.gnome.Platform.Locale/x86_64/3.22[$old]
XA: Transaction: update gnome:runtime/org.gnome.Platform.Locale/x86_64/3.24[$old]
XA: Transaction: update gnome:runtime/org.gnome.Platform/x86_64/3.22[$old]
XA: Transaction: install/update gnome:runtime/org.gnome.Platform.Locale/x86_64/3.22[/en]
XA: Transaction: update gnome:runtime/org.gnome.Platform/x86_64/3.24[$old]
XA: Transaction: install/update gnome:runtime/org.gnome.Platform.Locale/x86_64/3.24[/en]
XA: Transaction: update gnome:runtime/org.gnome.Sdk.Locale/x86_64/3.22[$old]
XA: Transaction: update gnome:runtime/org.gnome.Sdk.Locale/x86_64/3.24[$old]
XA: Transaction: update gnome:runtime/org.gnome.Sdk/x86_64/3.22[$old]
XA: Transaction: install/update gnome:runtime/org.gnome.Sdk.Locale/x86_64/3.22[/en]
XA: Transaction: update gnome:runtime/org.gnome.Sdk/x86_64/3.24[$old]
XA: Transaction: install/update gnome:runtime/org.gnome.Sdk.Locale/x86_64/3.24[/en]
XA: Transaction: update kde:runtime/org.kde.Platform.Locale/x86_64/master[$old]
XA: Transaction: update kde:runtime/org.kde.Platform/x86_64/master[$old]
XA: Transaction: install/update kde:runtime/org.kde.Platform.Locale/x86_64/master[/en]
XA: Transaction: update kde:runtime/org.kde.Sdk.Locale/x86_64/master[$old]                                                                                                                                                                                                       
XA: Transaction: update kde:runtime/org.kde.Sdk/x86_64/master[$old]                                                                                                                                                                                                              
XA: Transaction: install/update kde:runtime/org.kde.Sdk.Locale/x86_64/master[/en]                                                                                                                                                                                                
XA: Transaction: update org.telegram.desktop-origin:runtime/org.telegram.desktop.Locale/x86_64/master[$old]                                                                                                                                                                      
XA: Updating remote metadata for gnome                                                                                                                                                                                                                                           
XA: Updating remote metadata for gnome-apps                                                                                                                                                                                                                                      
XA: Updating remote metadata for kde                                                                                                                                                                                                                                             
XA: Updating remote metadata for kdeapps                                                                                                                                                                                                                                         
XA: Updating remote metadata for kdelocal                                                                                                                                                                                                                                        
XA: Ignoring automatic updates for system-helper remotes without gpg signatures                                                                                                                                                                                                  
XA: Updating remote metadata for kdew                                                                                                                                                                                                                                            
XA: Updating remote metadata for sekre                                                                                                                                                                                                                                           
XA: Ignoring automatic updates for system-helper remotes without gpg signatures                                                                                                                                                                                                  
XA: Updating remote metadata for org.kde.gcompris-origin                                                                                                                                                                                                                         
XA: Updating remote metadata for org.kde.katomic-1-origin                                                                                                                                                                                                                        
XA: Updating remote metadata for org.kde.katomic-10-origin                                                                                                                                                                                                                       
XA: Updating remote metadata for org.kde.katomic-11-origin                                                                                                                                                                                                                       
XA: Updating remote metadata for org.kde.katomic-12-origin                                                                                                                                                                                                                       
XA: Updating remote metadata for org.kde.katomic-2-origin                                                                                                                                                                                                                        
XA: Updating remote metadata for org.kde.katomic-3-origin                                                                                                                                                                                                                        
XA: Updating remote metadata for org.kde.katomic-4-origin                                                                                                                                                                                                                        
XA: Updating remote metadata for org.kde.katomic-5-origin
XA: Updating remote metadata for org.kde.katomic-6-origin
XA: Updating remote metadata for org.kde.katomic-7-origin
XA: Updating remote metadata for org.kde.katomic-8-origin
XA: Updating remote metadata for org.kde.katomic-9-origin
XA: Updating remote metadata for org.kde.katomic-origin
XA: Updating remote metadata for org.libreoffice.LibreOffice-origin
XA: Updating remote metadata for org.telegram.desktop-origin
Updating: org.gnome.Sdk/x86_64/3.24 from gnome
XA: starting fetch of ff76e78b238b407cf15f36469420c78b0697e881c16fd430187ee6a7662c09fe.commit (detached)
XA: fetch of ff76e78b238b407cf15f36469420c78b0697e881c16fd430187ee6a7662c09fe.commit (detached) complete
XA: starting fetch of ff76e78b238b407cf15f36469420c78b0697e881c16fd430187ee6a7662c09fe.commit
XA: fetch of ff76e78b238b407cf15f36469420c78b0697e881c16fd430187ee6a7662c09fe.commit complete
XA: write of ff76e78b238b407cf15f36469420c78b0697e881c16fd430187ee6a7662c09fe.commit complete
XA: starting fetch of ff76e78b238b407cf15f36469420c78b0697e881c16fd430187ee6a7662c09fe.commit (detached)
XA: fetch of ff76e78b238b407cf15f36469420c78b0697e881c16fd430187ee6a7662c09fe.commit (detached) complete
XA: pull: idle, exiting mainloop
[######=             ] Downloading: 65,5 kB/236,8 kB (65,5 kB/s)XA: fetch static delta part c9f631ecb5cf543bc9702d8ccb6775bcfd3c337f114f5321c4341d5d70cc8355 complete
XA: fetch static delta part 99cd64c5c11b7ac35a2830cf87978f39c374587ba1287e9d6b18f8375540378b complete
[#################=  ] Downloading: 207,7 kB/236,8 kB (103,8 kB/s)XA: fetch static delta part 859a169ec03121f13e761f4c00e6b4e903d33b445db1f22a077b028984b429b3 complete
XA: fetch static delta part c4f2fd0136f2f268ae88415787d152b4720be5cb95182efbea81e8a2ba312112 complete
[###################=] Downloading: 236,8 kB/236,8 kB (118,4 kB/s)XA: execute static delta part 99cd64c5c11b7ac35a2830cf87978f39c374587ba1287e9d6b18f8375540378b complete
XA: execute static delta part c4f2fd0136f2f268ae88415787d152b4720be5cb95182efbea81e8a2ba312112 complete
XA: execute static delta part c9f631ecb5cf543bc9702d8ccb6775bcfd3c337f114f5321c4341d5d70cc8355 complete
XA: execute static delta part 859a169ec03121f13e761f4c00e6b4e903d33b445db1f22a077b028984b429b3 complete
XA: pull: idle, exiting mainloop
[####################] 4 delta parts, 1 loose fetched; 231 KiB transferred in 3 secondsXA: Calling system helper: Deploy

Now at ff76e78b238b.
Updating: org.gnome.Sdk.Locale/x86_64/3.24 from gnome
XA: starting fetch of 9ad098593b2a098c896f6fa3af3715181aef96a0f8ade02120c6b169a4f4e027.commit (detached)
XA: fetch of 9ad098593b2a098c896f6fa3af3715181aef96a0f8ade02120c6b169a4f4e027.commit (detached) complete
XA: starting fetch of 9ad098593b2a098c896f6fa3af3715181aef96a0f8ade02120c6b169a4f4e027.commit
XA: fetch of 9ad098593b2a098c896f6fa3af3715181aef96a0f8ade02120c6b169a4f4e027.commit complete
XA: write of 9ad098593b2a098c896f6fa3af3715181aef96a0f8ade02120c6b169a4f4e027.commit complete
XA: starting fetch of 9ad098593b2a098c896f6fa3af3715181aef96a0f8ade02120c6b169a4f4e027.commit (detached)
XA: fetch of 9ad098593b2a098c896f6fa3af3715181aef96a0f8ade02120c6b169a4f4e027.commit (detached) complete
XA: pull: idle, exiting mainloop
XA: starting fetch of 9ad098593b2a098c896f6fa3af3715181aef96a0f8ade02120c6b169a4f4e027.commit (detached)
XA: starting fetch of 4e93ee19e83e9cbd45de07d71d4f80268ca6783c71c93638fba9ff2179611675.dirtree
[                    ] Downloading metadata: 0/(estimating) 0 bytes (0 bytes/s)XA: fetch of 9ad098593b2a098c896f6fa3af3715181aef96a0f8ade02120c6b169a4f4e027.commit (detached) complete
XA: fetch of 4e93ee19e83e9cbd45de07d71d4f80268ca6783c71c93638fba9ff2179611675.dirtree complete
XA: write of 4e93ee19e83e9cbd45de07d71d4f80268ca6783c71c93638fba9ff2179611675.dirtree complete
XA: starting fetch of e32488a9d97ea00b28d2c829d338b94190d679a5199ea0139cb97669c86e51c5.dirtree
[=                   ] Downloading metadata: 2/(estimating) 430 bytes (430 bytes/s)XA: fetch of e32488a9d97ea00b28d2c829d338b94190d679a5199ea0139cb97669c86e51c5.dirtree complete
XA: write of e32488a9d97ea00b28d2c829d338b94190d679a5199ea0139cb97669c86e51c5.dirtree complete
XA: pull: idle, exiting mainloop
[####################] 3 metadata, 0 content objects fetched; 13 KiB transferred in 1 secondsXA: Calling system helper: Deploy

Now at 9ad098593b2a.
Updating: org.freedesktop.Sdk/x86_64/1.6 from gnome
XA: starting fetch of e026259cd83125fe8b10009458f469f4f325dc13a4a8fd8c18a441f60b323a8e.commit (detached)
XA: fetch of e026259cd83125fe8b10009458f469f4f325dc13a4a8fd8c18a441f60b323a8e.commit (detached) complete
XA: starting fetch of e026259cd83125fe8b10009458f469f4f325dc13a4a8fd8c18a441f60b323a8e.commit
XA: fetch of e026259cd83125fe8b10009458f469f4f325dc13a4a8fd8c18a441f60b323a8e.commit complete
XA: write of e026259cd83125fe8b10009458f469f4f325dc13a4a8fd8c18a441f60b323a8e.commit complete
XA: starting fetch of e026259cd83125fe8b10009458f469f4f325dc13a4a8fd8c18a441f60b323a8e.commit (detached)
XA: fetch of e026259cd83125fe8b10009458f469f4f325dc13a4a8fd8c18a441f60b323a8e.commit (detached) complete
XA: pull: idle, exiting mainloop
[#################   ] Downloading: 81,9 kB/97,5 kB (81,9 kB/s)XA: fetch static delta part 3eb5760c07e2a0e7491d3091d7f6c049b56d45411d28ee1e5ad04a1a6fedc168 complete
XA: execute static delta part 3eb5760c07e2a0e7491d3091d7f6c049b56d45411d28ee1e5ad04a1a6fedc168 complete
XA: pull: idle, exiting mainloop
[####################] 1 delta parts, 1 loose fetched; 95 KiB transferred in 2 secondsXA: Calling system helper: Deploy

Now at e026259cd831.
Updating: org.gnome.Platform.Locale/x86_64/3.24 from gnome
XA: starting fetch of 46b68cc75f7319e304ecb71775be105459457df673c6189d61ad73325963a05c.commit (detached)
XA: fetch of 46b68cc75f7319e304ecb71775be105459457df673c6189d61ad73325963a05c.commit (detached) complete
XA: starting fetch of 46b68cc75f7319e304ecb71775be105459457df673c6189d61ad73325963a05c.commit
XA: fetch of 46b68cc75f7319e304ecb71775be105459457df673c6189d61ad73325963a05c.commit complete
XA: write of 46b68cc75f7319e304ecb71775be105459457df673c6189d61ad73325963a05c.commit complete
XA: starting fetch of 46b68cc75f7319e304ecb71775be105459457df673c6189d61ad73325963a05c.commit (detached)
XA: fetch of 46b68cc75f7319e304ecb71775be105459457df673c6189d61ad73325963a05c.commit (detached) complete
XA: pull: idle, exiting mainloop
XA: starting fetch of 46b68cc75f7319e304ecb71775be105459457df673c6189d61ad73325963a05c.commit (detached)
XA: starting fetch of 73dad277616d274846ce6417ee51a20361e065fe6e9e0103c8f598d2d8f6bd1c.dirtree
[                    ] Downloading metadata: 0/(estimating) 0 bytes (0 bytes/s)XA: fetch of 46b68cc75f7319e304ecb71775be105459457df673c6189d61ad73325963a05c.commit (detached) complete
XA: fetch of 73dad277616d274846ce6417ee51a20361e065fe6e9e0103c8f598d2d8f6bd1c.dirtree complete
XA: write of 73dad277616d274846ce6417ee51a20361e065fe6e9e0103c8f598d2d8f6bd1c.dirtree complete
XA: pull: idle, exiting mainloop
[####################] 2 metadata, 0 content objects fetched; 430 B transferred in 1 secondsXA: Calling system helper: Deploy

Now at 46b68cc75f73.
Updating: org.gnome.Platform/x86_64/3.24 from gnome
XA: starting fetch of 2e5369eb3dfae3f344e8181856cc3cb2c243401b4f9624f49425881b612983bc.commit (detached)
XA: fetch of 2e5369eb3dfae3f344e8181856cc3cb2c243401b4f9624f49425881b612983bc.commit (detached) complete
XA: starting fetch of 2e5369eb3dfae3f344e8181856cc3cb2c243401b4f9624f49425881b612983bc.commit
XA: fetch of 2e5369eb3dfae3f344e8181856cc3cb2c243401b4f9624f49425881b612983bc.commit complete
XA: write of 2e5369eb3dfae3f344e8181856cc3cb2c243401b4f9624f49425881b612983bc.commit complete
XA: starting fetch of 2e5369eb3dfae3f344e8181856cc3cb2c243401b4f9624f49425881b612983bc.commit (detached)
XA: fetch of 2e5369eb3dfae3f344e8181856cc3cb2c243401b4f9624f49425881b612983bc.commit (detached) complete
XA: pull: idle, exiting mainloop
[###=                ] Downloading: 16,4 kB/144,8 kB (16,4 kB/s)XA: fetch static delta part 4683e9da30f7b03a9d20a39fb3ca2f5e2cd996b72b6d024cc208237e87306e72 complete
[###############     ] Downloading: 104,8 kB/144,8 kB (52,4 kB/s)XA: fetch static delta part a79e285c61e120ace1e4eb63d63b334df95bbe5a11ceeca7a1057bb8412cbe9a complete
XA: fetch static delta part 64981d62a279bcc46d23cf5f5fdc69d494a11fa30b950c341ddc7fd80b6e50e3 complete
[##################  ] Downloading: 133,4 kB/144,8 kB (66,7 kB/s)XA: fetch static delta part 8570be61cf13c6a91372de51fd9a507a157834de3970cb7cc42e8d86ca60e1a5 complete
XA: execute static delta part 8570be61cf13c6a91372de51fd9a507a157834de3970cb7cc42e8d86ca60e1a5 complete
XA: execute static delta part a79e285c61e120ace1e4eb63d63b334df95bbe5a11ceeca7a1057bb8412cbe9a complete
[###################=] Downloading: 144,8 kB/144,8 kB (72,4 kB/s)XA: execute static delta part 4683e9da30f7b03a9d20a39fb3ca2f5e2cd996b72b6d024cc208237e87306e72 complete
XA: execute static delta part 64981d62a279bcc46d23cf5f5fdc69d494a11fa30b950c341ddc7fd80b6e50e3 complete
XA: pull: idle, exiting mainloop
[####################] 4 delta parts, 1 loose fetched; 141 KiB transferred in 2 secondsXA: Calling system helper: Deploy

Now at 2e5369eb3dfa.
Updating: org.kde.Sdk/x86_64/master from kde
Error: Failed to update org.kde.Sdk/x86_64/master: No such metadata object de367802b7af0e04e7d3a75ce7a2984d27ac3c05e1cc1ecca8ec9f5417f98910.commit
error: One or more operations failed
```

```
$ flatpak list 
Ref                                Options       
org.kde.Platform/arm/master        user,runtime  
org.kde.Sdk/arm/master             user,runtime  
org.gnome.Builder/x86_64/stable    system,current
org.gnome.Geary/x86_64/stable      system,current
org.kde.katomic/x86_64/master      system,current
org.kde.kdeconnect/x86_64/master   system,current
org.kde.kpat/x86_64/master         system,current
org.kde.kube/x86_64/master         system,current
org.telegram.desktop/x86_64/master system,current
org.freedesktop.Sdk/x86_64/1.6     system,runtime
org.gnome.Platform/x86_64/3.22     system,runtime
org.gnome.Platform/x86_64/3.24     system,runtime
org.gnome.Sdk/x86_64/3.22          system,runtime
org.gnome.Sdk/x86_64/3.24          system,runtime
org.kde.Platform/x86_64/master     system,runtime
org.kde.Sdk/x86_64/master          system,runtime
```

**Comment by aleixpol** [comment]
```
gnome                              Gnome Stable Runtimes               http://sdk.gnome.org/repo/                                            1        user                
gnome                              -                                   https://sdk.gnome.org/repo/                                           1        system              
gnome-apps                         -                                   https://sdk.gnome.org/repo-apps/                                      1        system              
kde                                -                                   http://distribute.kde.org/flatpak-testing/                            1        system              
kdeapps                            KDE Testing Applications            http://origin.distribute.kde.org/flatpak-apps-testing/                1        system              
kdelocal                           -                                   file:///home/apol/devel/playground/flatpak-kde-applications/repo      1        system,no-gpg-verify
kdew                               KDE Runtimes                        http://distribute.kde.org/flatpak-testing/                            1        system              
sekre                              -                                   file:///home/apol/projects/sekretario/flatpak/repo                    1        system,no-gpg-verify
org.kde.gcompris-origin            GCompris Educational Game           https://distribute.kde.org/flatpak-apps-testing/                      0        system,no-enumerate 
org.kde.katomic-1-origin           KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.kde.katomic-10-origin          KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.kde.katomic-11-origin          KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.kde.katomic-12-origin          KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.kde.katomic-2-origin           KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.kde.katomic-3-origin           KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.kde.katomic-4-origin           KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.kde.katomic-5-origin           KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.kde.katomic-6-origin           KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.kde.katomic-7-origin           KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.kde.katomic-8-origin           KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.kde.katomic-9-origin           KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.kde.katomic-origin             KAtomic                             https://distribute.kde.org/flatpak-testing/                           0        system,no-enumerate 
org.libreoffice.LibreOffice-origin The Document Foundation LibreOffice http://download.documentfoundation.org/libreoffice/flatpak/repository 0        system,no-enumerate 
org.telegram.desktop-origin        Telegram Desktop                    https://jgrulich.fedorapeople.org/telegram/repo/                      0        system,no-enumerate 
```

**Comment by alexlarsson** [comment]
This error:
```
Updating: org.kde.Sdk/x86_64/master from kde
Error: Failed to update org.kde.Sdk/x86_64/master: No such metadata object de367802b7af0e04e7d3a75ce7a2984d27ac3c05e1cc1ecca8ec9f5417f98910.commit
```
Means that ostree was somehow failing to find this file:
```
http://distribute.kde.org/flatpak-testing/objects/de/367802b7af0e04e7d3a75ce7a2984d27ac3c05e1cc1ecca8ec9f5417f98910.commit
```

Do you have access to the server logs? Did it try to load the file and get some kind of error?

**Comment by alexlarsson** [comment]
Actually, that is not true... That error is when the file does not exist locally. Hmm....

**Comment by alexlarsson** [comment]
You don't happen to be running some update operations in parallel? Like from e.g the kde software thing in the background?

**Comment by alexlarsson** [comment]
Here is what seems to be happening:

We decide to update org.kde.Sdk/x86_64/master, among others. We then update the others first, and when we get to it, we try to load the current commit in the local repo, which we do because we want to compare the old and new commit timestamp (not allowing downgrades).

We do this by reading the ref file for the /var/lib/flatpak repo (in this case something like 
`/var/lib/flatpak/repo/refs/remotes/kde/runtime/org.kde.Sdk/x86_64/master`). This file contains the commit id that branch is currently at, we then look in `/var/lib/flatpak/repo/objects/` for this commit, but it doesn't exist. I have no idea why this would happen though.

Just to make sure, can you try cleaning out  `~/.cache/flatpak/` and see if that changes anything.

**Comment by aleixpol** [comment]
```
$ rm -rf .cache/flatpak/system-cache/
$ flatpak update
Looking for updates...
Updating: org.kde.Sdk/x86_64/master from kde
Error: Failed to update org.kde.Sdk/x86_64/master: No such metadata object de367802b7af0e04e7d3a75ce7a2984d27ac3c05e1cc1ecca8ec9f5417f98910.commit
error: One or more operations failed
```

**Comment by olantwin** [comment]
I also seem to have run into this (maybe having another case might narrow down what's happening?):

```
$ flatpak update -v --ostree-verbose
F: Updating remote metadata for flathub
F: Updating remote metadata for org.mozilla.FirefoxRepo
Updating appstream data for remote flathub
F: Reusing tmpdir staging-71ed55b8-d881-467b-87d8-c53fa8050f62-aNSVVw
Error updating: No such metadata object 85891bd89c0bd2a74b81b2d4d327ab4ab0f3951de86d258ca8202c2b758b4a76.commit
Updating appstream data for remote org.mozilla.FirefoxRepo
F: Reusing tmpdir staging-71ed55b8-d881-467b-87d8-c53fa8050f62-aNSVVw
Error updating: No such metadata object 8414748774145985cc1dafc73f88825ba769fcffb9e979018e47eaf7069de55a.commit
```
This is after `rm -rf .cache/flatpak/system-cache/`.

Is there any way to clear the relevant state?

**Comment by aleixpol** [comment]
FWIW, I haven't seen this issue in months.

**Comment by olantwin** [comment]
It seems to come and go for me (on the timescale of weeks). No pattern I can determine so far.

**Comment by Pointedstick** [comment]
I'm seeing it right now, and can reproduce on demand, if folks need me to do some debugging

```
$  sudo flatpak update
[sudo] password for nate: 
Looking for updates...
Updating: org.freedesktop.Platform.Locale/x86_64/1.6 from flathub
No updates.
Updating: org.freedesktop.Platform/x86_64/1.6 from flathub
No updates.
Installing: org.freedesktop.Platform.ffmpeg/x86_64/1.6 from flathub

1 delta parts, 3 loose fetched; 2186 KiB transferred in 3 seconds                         
Updating: org.gnome.Platform.Locale/x86_64/3.24 from flathub


Warning: Failed to update org.gnome.Platform.Locale/x86_64/3.24: While pulling runtime/org.gnome.Platform.Locale/x86_64/3.24 from remote flathub: No such metadata object 4f9099920477c6784621cc87af5a90648fd5a01b00b3e9270e2ebbffee85b2f6.commit
Updating: org.gnome.Platform.Locale/x86_64/3.26 from flathub


Warning: Failed to update org.gnome.Platform.Locale/x86_64/3.26: While pulling runtime/org.gnome.Platform.Locale/x86_64/3.26 from remote flathub: No such metadata object 5b27b4a6d6e2247a6b27e5b7d95133f5cc52159136e418fe42485a0635d38c69.commit
Updating: org.gnome.Platform/x86_64/3.24 from flathub


Error: Failed to update org.gnome.Platform/x86_64/3.24: While pulling runtime/org.gnome.Platform/x86_64/3.24 from remote flathub: No such metadata object 4336f092c117d3eacfdbaacc8996e94786e55b8d6a93cece00fef39648b298d0.commit
Updating: org.gnome.Platform/x86_64/3.26 from flathub


Error: Failed to update org.gnome.Platform/x86_64/3.26: While pulling runtime/org.gnome.Platform/x86_64/3.26 from remote flathub: No such metadata object c4d67937b0417cb0a83b7a836b0c2052b9e94bcc3978847f2a262fa8c1343c50.commit
error: One or more operations failed

```

**Comment by limkokhole** [comment]
I just encounter "no such metadata object" and "failed to read commit" errors in Ubuntu 17.10 when click update "GNOME Application Platform version 3.26 " via gnome-software gui.

I pasted [my /var/log/syslog in pastebin](https://pastebin.com/miUMsuib), just in case it may help.

**Comment by Pointedstick** [comment]
This issue is preventing me from receiving any updates for for my Flatpak apps that use the GNOME and FreeDesktop runtimes. Is there any workaround, at least?

Here' the log output from `flatpak update -v --ostree-verbose` and `flatpak remotes -d`:

[Log output.txt](https://github.com/flatpak/flatpak/files/2127661/Log.output.txt)

**Comment by TJProgrammer** [comment]
Hello!

I've the same issue when I try to install Spotify.
I tried run `update` and I got this:
```
[root@localhost ~]# flatpak update -v
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /root/.local/share/flatpak
F: No installations directory in /etc/flatpak/installations.d. Skipping
Looking for updates...
F: Fetching summary file for remote ‘flathub’
F: Failed to download optional summary
F: Transaction: update flathub:runtime/org.freedesktop.Platform.Locale/x86_64/18.08[$old]
F: Transaction: update flathub:runtime/org.freedesktop.Platform.VAAPI.Intel/x86_64/18.08[$old]
F: Transaction: update flathub:runtime/org.freedesktop.Platform.html5-codecs/x86_64/18.08[$old]
F: Transaction: update flathub:runtime/org.freedesktop.Platform/x86_64/18.08[$old]
F: Updating remote metadata for flathub
F: Fetching summary file for remote ‘flathub’
flatpak update: Error updating remote metadata for 'flathub': [35] SSL connect error
F: Fetching summary file for remote ‘flathub’
F: Failed to download optional summary
flatpak update: Warning: Treating remote fetch error as non-fatal since runtime/org.freedesktop.Platform/x86_64/18.08 is already installed: Unable to load summary from remote flathub: [35] SSL connect error
flatpak update: Warning: Can't find runtime/org.freedesktop.Platform/x86_64/18.08 metadata for dependencies: Unable to load metadata from remote flathub: summary fetch error: [35] SSL connect error
F: marking op update:runtime/org.freedesktop.Platform/x86_64/18.08 resolved to 95881b6e72a7daf67c1ffea66df151c23ef70a36806196081ea7f7e062db7e78
flatpak update: Warning: Treating remote fetch error as non-fatal since runtime/org.freedesktop.Platform.html5-codecs/x86_64/18.08 is already installed: Unable to load summary from remote flathub: [35] SSL connect error
flatpak update: Warning: Can't find runtime/org.freedesktop.Platform.html5-codecs/x86_64/18.08 metadata for dependencies: Unable to load metadata from remote flathub: summary fetch error: [35] SSL connect error
F: marking op update:runtime/org.freedesktop.Platform.html5-codecs/x86_64/18.08 resolved to 0e0cea6e038f71f49091db5cbd2186d1d679a299913897e0a11f6d87b038539a
flatpak update: Warning: Treating remote fetch error as non-fatal since runtime/org.freedesktop.Platform.VAAPI.Intel/x86_64/18.08 is already installed: Unable to load summary from remote flathub: [35] SSL connect error
flatpak update: Warning: Can't find runtime/org.freedesktop.Platform.VAAPI.Intel/x86_64/18.08 metadata for dependencies: Unable to load metadata from remote flathub: summary fetch error: [35] SSL connect error
F: marking op update:runtime/org.freedesktop.Platform.VAAPI.Intel/x86_64/18.08 resolved to 041424177ff848628c84479e65e5d3ca9fc73702868c8d608e02f0ce99f444ae
flatpak update: Warning: Treating remote fetch error as non-fatal since runtime/org.freedesktop.Platform.Locale/x86_64/18.08 is already installed: Unable to load summary from remote flathub: [35] SSL connect error
flatpak update: Warning: Can't find runtime/org.freedesktop.Platform.Locale/x86_64/18.08 metadata for dependencies: Unable to load metadata from remote flathub: summary fetch error: [35] SSL connect error
F: marking op update:runtime/org.freedesktop.Platform.Locale/x86_64/18.08 resolved to be0744bd1e50fa5ec0ec6c3cd5652f8b4bd48686cebabf1975cfa5c928b06755
F: no resolved metadata for related to runtime/org.freedesktop.Platform/x86_64/18.08
F: no resolved metadata for related to runtime/org.freedesktop.Platform.html5-codecs/x86_64/18.08
F: no resolved metadata for related to runtime/org.freedesktop.Platform.VAAPI.Intel/x86_64/18.08
F: no resolved metadata for related to runtime/org.freedesktop.Platform.Locale/x86_64/18.08
F: flathub:x86_64 appstream age 739 is less than ttl 86400
```

OS information : fedora 29
`Linux localhost.localdomain 4.18.16-300.fc29.x86_64 #1 SMP Sat Oct 20 23:24:08 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`

Flatpak version: `Flatpak 1.0.4`

**Comment by scx** [comment]
I had a similar problem.

```
$ flatpak update
Looking for updates...
Updating in system:
org.freedesktop.Platform/i386/18.08               flathub 85a7cf4d95e2
com.github.artemanufrij.regextester/x86_64/stable flathub a36327f01ff7
org.gnome.Platform/x86_64/3.30                    flathub f6b4bc140bb2
org.gnome.Sdk/x86_64/3.28                         flathub 31af0c10f8e4
org.kde.Platform.Locale/x86_64/5.11               flathub 67caec848090
org.kde.Platform/x86_64/5.9                       flathub 606e75d0708e
Is this ok [y/n]: y
Updating: org.freedesktop.Platform/i386/18.08 from flathub
Error: Failed to update org.freedesktop.Platform/i386/18.08: No such metadata object 15e017088a9b5c8db86a1d246bfcdb7358e6b6b37f254555709dfbb5516fa54f.commit
Updating: com.github.artemanufrij.regextester/x86_64/stable from flathub
Error: Failed to update com.github.artemanufrij.regextester/x86_64/stable: No such metadata object 9618e34d742c907e1a0001b78ed005070ff6fa83938310ea808f320a54c49784.commit
Updating: org.gnome.Platform/x86_64/3.30 from flathub
Error: Failed to update org.gnome.Platform/x86_64/3.30: No such metadata object 124b2831f375c32e087583b2b86ddd9f15fa107ed210cad95abf768a476c3f9c.commit
Updating: org.gnome.Sdk/x86_64/3.28 from flathub
Error: Failed to update org.gnome.Sdk/x86_64/3.28: No such metadata object 70d9a1d6966acf73797ea555322422f62220b7bf5f938473d46c3565ad8e1518.commit
Updating: org.kde.Platform.Locale/x86_64/5.11 from flathub
Warning: Failed to update org.kde.Platform.Locale/x86_64/5.11: No such metadata object e764eeecd976b343a76b6fb33cc056b48cc6fa0c44634c916a39c3b4bbbf2599.commit
Updating: org.kde.Platform/x86_64/5.9 from flathub
Error: Failed to update org.kde.Platform/x86_64/5.9: No such metadata object 950ba90bd04cd2a34ca546a7cc587959425f184dca80bbfcba67b82bbe4a021d.commit
error: There were one or more errors
```

Try running `flatpak repair` with root privileges.
It [worked](https://pastebin.com/raw/fGqBRBn9) for me.

**Comment by GetCurious** [comment]
> I had a similar problem.
> Try running `flatpak repair` with root privileges.
> It [worked](https://pastebin.com/raw/fGqBRBn9) for me.

Thanks! linux newbie here, flatpak repair worked

**Comment by afcady** [comment]
flatpak repair fixed it for me too.  Can at least the error message be changed to suggest trying this?

**Comment by cobalt2727** [comment]
The error message absolutely should be changed to suggest a `sudo flatpak repair`, worked perfectly for me

**Comment by claudio-walser** [comment]
I don't understand why this issue is closed. This problem clearly isn't solved yet. I had it right now on Fedora 35.
Or is it intended to be documented as github issue? A modified error message would already help a lot and preventing thousand of peoples wasting their time by searching for the solution.

**Comment by tjapro** [comment]
Sorry, @matthiasclasen , but why you closed this issue, even with the above requests?

Can you give some explanation? Thanks!

**Comment by mwleeds** [comment]
This might be a duplicate of #3479, hard to say for sure. If anyone sees an issue like this, please open a new bug report rather than commenting here, and include the version of Flatpak you're on, and ideally the output of `ostree --version` as well.

**Comment by tjapro** [comment]
> This might be a duplicate of #3479, hard to say for sure. If anyone sees an issue like this, please open a new bug report rather than commenting here, and include the version of Flatpak you're on, and ideally the output of `ostree --version` as well.

Thanks for the explanation! Just a hint: close this issue to comments.
